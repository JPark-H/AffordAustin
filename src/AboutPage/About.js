import './About.css';
import { Container,Button, Alert, Card, Row, Col, ListGroup } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css'
import React, { useEffect, useState } from "react";
//import * as Icon from 'bootstrap-icons/font/bootstrap-icons.css';
import * as Icon from 'react-bootstrap-icons';
import { IconRotate_90DegreesCcw } from '@aws-amplify/ui-react';
const Koala = "https://res.cloudinary.com/teepublic/image/private/s--xY-2bGLB--/t_Resized%20Artwork/c_fit,g_north_west,h_1054,w_1054/co_ffffff,e_outline:53/co_ffffff,e_outline:inner_fill:53/co_bbbbbb,e_outline:3:1000/c_mpad,g_center,h_1260,w_1260/b_rgb:eeeeee/c_limit,f_auto,h_630,q_90,w_630/v1591112325/production/designs/10907156_0.jpg";
const teamInfo = [
	{
		name: "Koala 1",
		// username: "The Krusty Krab",
		// email: "knees weak",
		picture: Koala,
		role: "Sleeping",
		bio:
			"The Krusty Krab Pizza, is the Pizza, for you and me!",
		commits: 1192,
		issues: 0,
		tests: 0,
	},
	{
		name: "Koala 2",
		// username: "Bikini Bottom",
		// email: "arm's spaghetti",
		picture: Koala,
		role: "Hibernating",
		bio:
			"The fitness gram pacer test is a multi-stage aerobic ",
		commits: 0,
		issues: 0,
		tests: 271,
	},
	{
		name: "Koala 3",
		// username: "Goofy Goobers",
		// email: "Sakurajima Mai <3",
		picture: Koala,
		role: "Drinking",
		bio:
			"Have ",
		commits: 0,
		issues: 0,
		tests: 0,
	},
	{
		name: "Koala 4",
		// username: "Salty Spitoon",
		// email: "Lorem Ipsum",
		picture: Koala,
		role: "Breathing",
		bio:
			"My paws can't even reach the keyboard man",
		commits: 0,
		issues: 69,
		tests: 0,
	},
	{
		name: "Koala 5",
		// username: "Weenie Hut Jr",
		// email: "Inconceivable",
		picture: Koala,
		role: "Crying",
		bio: 
      "Do Koalas even have claws?",
		commits: 0,
		issues: 0,
		tests: 123,
	}
]



function MemberCards() {
  const [show, setShow] = useState(true);

  return (

	  <Container>
	  <><Row className="g-5 m-1 justify-content-center"  xs="auto" style={{width: "100%"}}>
      {teamInfo.slice(0, 5).map((teamMember) => {
        const { name, username, picture, email, role, bio, commits, issues, tests } = teamMember;
        return (
			<Col xs="auto" key={teamMember}>
				<Card style={{ width: '100%' }}>
					<Card.Img variant="top" src={picture}
						className="about_card rounded-circle mb-0 mt-3 mx-auto" style={{ width: '9em', height: '9em'}} />
					<Card.Body className = "about_info" style={{ backgroundColor: "white", color: "black", width: "10em" }}>
						<Card.Title className="mt-1 card_names">
							<strong>{name}</strong>
						</Card.Title>
						<Card.Subtitle className="role m-2 text-muted" style={{fontSize:"100%"}}>{role}</Card.Subtitle>
						<p className="about_text">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
						<hr className="mb-3 m-2" style={ {height:"2px"} }/>
						<Row>
							<Col className="stats">
								<p className="stat_text">Commits</p>
								<Icon.Check2Circle style={{color: "black"}}></Icon.Check2Circle>
								<p>{commits}</p>
							</Col>
							<Col className="stats">
								<p className="stat_text">Issues</p>
								<Icon.ListCheck></Icon.ListCheck>
								<p>{issues}</p>
							</Col>
							<Col className="tests">
								<p className="stat_text">Tests</p>
								<Icon.Wrench style={{color:"black"}}></Icon.Wrench>
								<p>{tests}</p>
							</Col>
						</Row>

					</Card.Body>
				</Card>
			</Col>
        );
      })}
      </Row></>
	  </Container>

  );
}

function About() {
  return (
    <div className="App" style = {{backgroundColor:"#282c34", paddingBottom:"3em"}}>
      <header className="App-header">
      </header>
	  <MemberCards />
    </div>
  );
}

export default About;
