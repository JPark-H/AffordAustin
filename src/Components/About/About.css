.App {
  text-align: center;
}

.section_header {
  color: black;
  font-size: max(1em, 40px);
  font-weight: bolder;
  padding-top: 1em;
  margin-bottom: .5em;
}

.temp_interesting {
  color: black;
  font-size: max(.9em, 30px);
  margin-bottom: .5em;
}

.tools_apis {
  width: 85%;
  font-size: 28px;
}

.instance_data {
  width: 85%;
  font-size: 28px;
}

.about_text {
  font-size: max(.7em, 24px);
  width: 95%;
}

.link b {
  color: blue;
}

.list-group-item {
  margin-bottom: 10px;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
}