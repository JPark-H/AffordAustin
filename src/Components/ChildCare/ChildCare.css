.info_page {
    align-items: center;
    justify-content: center;
    margin-top: .7em;
    min-width: 350px;
    min-height: 100vh;
    background-color: #f0f2f5;
    width: 80%;
  }
  
  .instance_header {
    font-weight: 500;
    color: black;
    text-align: center;
    font-size: max(1.4em, 38px);
  }
  
  .instance_info {
    max-width: 1000px;
    min-height: 700px;
    background-color: white;
    color: black;
    margin-bottom: 3rem;
    padding-bottom: 3rem;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  }
  
  .attribute_head {
    padding-top: .5em;
    font-size: 28px;
  }
  
  .c_image {
    width: 75%;
    margin-bottom: 1em;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  }
  
  .c_map iframe {
    width: 95%;
    height: 18em;
    margin-bottom: 1em;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  }
  
  .instance_data {
    width: 85%;
    font-size: 28px;
  }
  
  .attribute_list {
    font-size: max(.7em, 18px);
  }
  
  .instance_connect {
    width: 100%;

  }