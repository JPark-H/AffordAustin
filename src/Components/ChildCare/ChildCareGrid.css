.child_care_grid {
    align-items: center;
    justify-content: center;
    text-align: left;
    white-space: pre-wrap;
    color: black;  
    min-height: 100vh;
    width: 100vw;
    margin-bottom: 0;
}

.header {
    font-weight: 500;
    font-size: 50px;
    color: black;
    text-align: center;
    margin: 0;
  }

.inst_card {
    font-size: 15px;
    width: 350px;
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
}


a {
    color: black;
    text-decoration: none;
}