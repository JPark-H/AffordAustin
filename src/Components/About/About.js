import { Container, Button, Alert, Card, Row, Col, ListGroup } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css'
import React, { useEffect, useState } from "react";
//import * as Icon from 'bootstrap-icons/font/bootstrap-icons.css';
import * as Icon from 'react-bootstrap-icons';
import { IconRotate_90DegreesCcw } from '@aws-amplify/ui-react';
import teamInfo from "./TeamData"
import './About.css'
function MemberCards() {
	return (
		<Container>
			<><Row className="g-5 m-0 justify-content-center" xs="auto" style={{ width: "100%" }}>
				{teamInfo.slice(0, 5).map((teamMember) => {
					const { name, username, picture, email, role, bio, commits, issues, tests } = teamMember;
					return (
						<Col xs="auto" key={teamMember}>
							<Card style={{ width: '100%', borderRadius:"2rem"}}>
								<Card.Img variant="top" src={picture}
									className="about_card circle-rounded mb-0 mt-4 mx-auto" style = {{borderRadius:"2rem"}} />
								<Card.Body className="about_info" style={{ backgroundColor: "white", color: "black", width: "10em", borderRadius:"3rem" }}>
									<Card.Title className="mt-1 card_names">
										<strong>{name}</strong>
									</Card.Title>
									<Card.Subtitle className="role m-2 text-muted" style={{ fontSize: "100%" }}>{role}</Card.Subtitle>
									<p className="about_text">
										asdsum dolor sit amet, consectetur adipiscing elit, sed do 
										eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
										exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in 
										reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat 
										cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
									</p>
									<hr className="mb-3 m-2" style={{ height: "2px" }} />
									<Row>
										<Col className="stats">
											<p className="stat_text">Commits</p>
											<Icon.Check2Circle style={{ color: "black" }}></Icon.Check2Circle>
											<p className="stat_num">{commits}</p>
										</Col>
										<Col className="stats">
											<p className="stat_text">Issues</p>
											<Icon.ListCheck></Icon.ListCheck>
											<p className="stat_num">{issues}</p>
										</Col>
										<Col className="tests">
											<p className="stat_text">Tests</p>
											<Icon.Wrench></Icon.Wrench>
											<p className="stat_num">{tests}</p>
										</Col>
									</Row>

								</Card.Body>
							</Card>
						</Col>
					);
				})}
			</Row></>
		</Container>
	);
}

function About() {
	return (
		<div className="App" style={{ backgroundColor: "#282c34", paddingBottom: "3em" }}>
			<MemberCards />
			<MemberCards />
		</div>
	);
}

export default About;
