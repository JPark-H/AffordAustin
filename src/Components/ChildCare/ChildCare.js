import './ChildCare.css';
import 'bootstrap/dist/css/bootstrap.min.css'

import { Container, Row, Col, Image, ListGroup, Card, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const ChildCare = ({ child_care }) => {
    return (
        <div style={{ backgroundColor: "#f0f2f5" }}>
            <div className="info_page mx-auto justify-content-center">
                <p className="instance_header">{child_care.operation_name}</p>
                <Container className="instance_info">
                    <p className="attribute_head">Location Image</p>
                    <Image className="h_image" rounded src={child_care.img}></Image>
                    <Row>
                        <Col>
                            <Card className="instance_data mx-auto" style={{
                                borderTopRightRadius: "2rem",
                                borderTopLeftRadius: "2rem"
                            }}>
                                <Card.Header style={{
                                    borderTopRightRadius: "2rem",
                                    borderTopLeftRadius: "2rem"
                                }}>Data</Card.Header>
                                <ListGroup className="attribute_list" style={{ textAlign: "left" }} variant="flush">
                                    <ListGroup.Item><b>Operation Type:</b> {child_care.operation_type}</ListGroup.Item>
                                    <ListGroup.Item><b>Programs Provided:</b> {child_care.programs_provided}</ListGroup.Item>
                                    <ListGroup.Item><b>Administrator's Name:</b> {child_care.administrator_director_name}</ListGroup.Item>
                                    <ListGroup.Item><b>Accepts Child Care Subsidies:</b> {child_care.accepts_child_care_subsidies}</ListGroup.Item>
                                    <ListGroup.Item><b>Days of Operation:</b> {child_care.days_of_operation}</ListGroup.Item>
                                    <ListGroup.Item><b>Hours of Operations:</b> {child_care.hours_of_operation}</ListGroup.Item>
                                    <ListGroup.Item><b>Licensed to Serve Ages:</b> {child_care.licensed_to_serve_ages}</ListGroup.Item>
                                </ListGroup>
                            </Card>
                        </Col>
                        <Col>
                            <Card className="instance_data mx-auto" style={{
                                borderTopRightRadius: "2rem",
                                borderTopLeftRadius: "2rem"
                            }}>
                                <Card.Header style={{
                                    borderTopRightRadius: "2rem",
                                    borderTopLeftRadius: "2rem"
                                }}>Contact Information</Card.Header>
                                <ListGroup className="attribute_list" style={{ textAlign: "left" }} variant="flush">
                                    <ListGroup.Item><b>Phone Number:</b> {child_care.phone_number}</ListGroup.Item>
                                    <ListGroup.Item><b>Website:</b> <a href={child_care.website_address}>{child_care.website_address}</a></ListGroup.Item>
                                    <ListGroup.Item><b>Address:</b> {child_care.location_address}</ListGroup.Item>
                                    <ListGroup.Item><b>Mailing Address:</b> {child_care.mailing_address}</ListGroup.Item>
                                    <ListGroup.Item><b>County:</b> {child_care.county}</ListGroup.Item>
                                    <ListGroup.Item><b>Email Address:</b> {child_care.email_address}</ListGroup.Item>
                                </ListGroup>
                            </Card>
                        </Col>
                    </Row>
                    <p className="attribute_head">Map</p>
                    <div className='h_map' dangerouslySetInnerHTML={{ __html: child_care.map }}></div>
                    <Row className="mt-3 justify-content-between mx-auto" >
                        <Link to='/Jobs'>
                        <Col className="mx-auto"><Button variant="secondary">Find Nearby Jobs</Button>{' '}</Col>
                        </Link>
                        <Link to='/Housing'>
                        <Col className="mx-auto"><Button variant="secondary">Find Nearby Housing</Button>{' '}</Col>
                        </Link>
                    </Row>
                </Container>
            </div>
        </div>
    );
};

const ChildCare1 = () => {
    const child_care = {
        "id":"1",
        "operation_type":"Licensed Center",
        "operation_name":"Zilker EAC YMCA",
        "programs_provided":"School Age Care, Children with Special Needs, After School Care, Snacks Provided, Field Trips",
        "location_address":"1900 Bluebonnet Ln, Austin TX, 78704",
        "mailing_address":"55 N INTERSTATE 35, Austin TX 78702",
        "phone_number":"(512) 236-9622",
        "county":"TRAVIS",
        "website_address":"www.eacymca.org",
        "administrator_director_name":"Michelle Torres",
        "accepts_child_care_subsidies":"Yes",
        "hours_of_operation":"02:45 PM - 06:30 PM",
        "days_of_operation":"Monday-Friday",
        "licensed_to_serve_ages":"Pre-Kindergarten, School",
        "total_capacity":"85",
        "location_address_geo":{
            "latitude":"30.252383",
            "longitude":"-97.773635",
            "human_address":"{\"address\": \"1900 BLUEBONNET LN AUSTIN TX- 78704 3342\", \"city\": \"\", \"state\": \"\", \"zip\": \"\"}"
        },
        "email_address":"none",
        "img":"https://www.austinymca.org/themes/custom/www_bootstrap/logo.svg",
        "map":'<iframe src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d6892.914952294049!2d-97.774567!3d30.252544000000004!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x8644b527df4e3533%3A0xb48340c3315e1f89!2s1900%20Bluebonnet%20Ln%2C%20Austin%2C%20TX%2078704!5e0!3m2!1sen!2sus!4v1646089221646!5m2!1sen!2sus" width="600" height="450" style="border:0;" allowfullscreen="" loading="lazy"></iframe>'
    };

    return (
        <ChildCare child_care={ child_care }/>
    )
}

const ChildCare2 = () => {
    const child_care = {
        "id":"2",
        "operation_type":"Licensed Center",
        "operation_name":"Childrens Center of Austin",
        "programs_provided":"Meals Provided, After School Care, Snacks Provided, Drop-In Care, Skill Classes, Part Time Care, Transportation to/from School, Field Trips",
        "location_address":"6507 Jester Blvd Bldg II, Austin TX, 78750",
        "mailing_address":"6507 Jester Blvd Bldg II, Austin TX, 78750",
        "phone_number":"(512) 795-8300",
        "county":"TRAVIS",
        "website_address":"www.childrenscenterofaustin.com",
        "administrator_director_name":"Erica Gonzales",
        "accepts_child_care_subsidies":"No",
        "hours_of_operation":"06:30 AM-06:00 PM",
        "days_of_operation":"Monday-Friday",
        "total_capacity":"275",
        "licensed_to_serve_ages":"Infant, Toddler, Pre-Kindergarten, School",
        "location_address_geo":{
            "latitude":"30.37021",
            "longitude":"-97.800817",
            "human_address":"{\"address\": \"6507 JESTER BLVD BLDG II AUSTIN TX- 78750 8368\", \"city\": \"\", \"state\": \"\", \"zip\": \"\"}"
        },
        "email_address":"jesteradmin@childrenscenterofaustin.com",
        "img": "https://wp02-media.cdn.ihealthspot.com/wp-content/uploads/sites/567/2019/12/09185356/logo.png",
        "map":'<iframe src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d6884.679779894778!2d-97.799367!3d30.369704999999996!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x865b34b96502189f%3A0x974c7afd1bdf0293!2s6507%20Jester%20Blvd%2C%20Austin%2C%20TX%2078750!5e0!3m2!1sen!2sus!4v1646089350104!5m2!1sen!2sus" width="600" height="450" style="border:0;" allowfullscreen="" loading="lazy"></iframe>'
    };

    return (
        <ChildCare child_care={ child_care }/>
    )
}

const ChildCare3 = () => {
    const child_care = {
        "id":"3",
        "operation_type":"Licensed Child-Care Home",
        "operation_name":"A+ Kids Playschool",
        "programs_provided":"Meals Provided, Snacks Provided",
        "location_address":"17257 Tobermory Dr, Pflugerville TX, 78660",
        "mailing_address":"17257 Tobermory Dr, Pflugerville TX, 78660",
        "phone_number":"(512) 989-5977",
        "county":"TRAVIS",
        "website_address":"www.aplusdaycare.com",
        "administrator_director_name":"Damita Taymullah",
        "accepts_child_care_subsidies":"No",
        "hours_of_operation":"07:00 AM-05:30 PM",
        "days_of_operation":"Monday-Friday",
        "total_capacity":"12",
        "licensed_to_serve_ages":"Infant, Toddler, Pre-Kindergarten, School",
        "location_address_geo":{
            "latitude":"30.46847",
            "longitude":"-97.639908",
            "human_address":"{\"address\": \"17257 TOBERMORY DR PFLUGERVILLE TX- 78660 1726\", \"city\": \"\", \"state\": \"\", \"zip\": \"\"}"
        },
        "email_address":"none",
        "img":"https://static.wixstatic.com/media/c71ae4_5e977ff378404fb5a58ef32a74e1a11b~mv2_d_3024_4032_s_4_2.jpg/v1/fill/w_514,h_685,al_c,q_80,usm_0.66_1.00_0.01/c71ae4_5e977ff378404fb5a58ef32a74e1a11b~mv2_d_3024_4032_s_4_2.webp",
        "map":'<iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3438.8636644116596!2d-97.64232048503123!3d30.468299481730075!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x8644cfcfa0645413%3A0xe805c658d49d7710!2sA%2B%20Kids%20Playschool!5e0!3m2!1sen!2sus!4v1646089433674!5m2!1sen!2sus" width="600" height="450" style="border:0;" allowfullscreen="" loading="lazy"></iframe>'
    };

    return (
        <ChildCare child_care={ child_care }/>
    )
}

export { ChildCare1, ChildCare2, ChildCare3 };