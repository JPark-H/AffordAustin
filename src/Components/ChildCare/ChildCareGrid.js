import './ChildCareGrid.css'
import 'bootstrap/dist/css/bootstrap.min.css'

import { Container, Card, Row, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { useState, useEffect } from 'react';
import Paginate from './../Pagination/Pagination';
import axios from 'axios';
import Koala from './../About/MemberCards/imgs/Koallaaaaa.png'

const ChildCareGrid = () => {
    const [programs, setPrograms] = useState([]);
    const [loading, setLoading] = useState(true);
    const [currentPage, setCurrentPage] = useState(1);
    const [totalNumPrograms, setTotalNumPrograms] = useState(1);
    const [programsPerPage, setProgramsPerPage] = useState(21);

    const getChildCareData = async (page, query) => {
        setLoading(true);
        axios.defaults.headers.common['Content-Type'] = 'application/vnd.api+json'
        axios.defaults.headers.common['Accept'] = 'application/vnd.api+json'
        const endpoint = `http://localhost:5000/api/childcare?page[size]=${programsPerPage}&page[number]=${page}`;
        // const endpoint = `http://api.affordaustin.me/api/childcare?page[size]=${programsPerPage}&page[number]=${page}`;
        const data = await axios.get(endpoint);
        setTotalNumPrograms(data.data.meta.total);
        setPrograms(data.data.data);
        setLoading(false);
    }

    useEffect(() => {
        getChildCareData(1, '');
    }, []);

    const paginate = (pageNum) => {
        setCurrentPage(pageNum);
        getChildCareData(pageNum, '');
    }

    return (
        <div style={{ backgroundColor: "#f0f2f5" }}>
            <div className='child_care_grid mx-auto'>
                <Container fluid>
                    <Row>
                        <Col className='header'>Child Care</Col>
                    </Row>
                    <Row>
                        <Paginate totalInstances={totalNumPrograms} pageLimit={programsPerPage} paginate={paginate} />
                    </Row>
                    <h1 style = {{fontSize:"40px", textAlign:"center"}}>Showing {programs.length} Results</h1>
                    <Row className="g-3 justify-content-center" xs='auto'>
                        {loading ? <h3>Loading</h3> : programs.map(program => {
                            return (
                            <Col key={program.id}>
                                <InstanceCard child_care={program.attributes} id={program.id}/>
                            </Col>);
                        })}
                    </Row>
                </Container>
            </div>
        </div>
    );
}

const InstanceCard = ({ child_care, id }) => {
    const link = `/ChildCare/${ id }`;
    let ages = child_care.licensed_to_serve_ages.replaceAll(",", ", ");
    return (
        <Link to={ link }>
            <Card className='c_inst_card'>
                {/* Replace */}
                <Card.Img variant='top' src={Koala} />
                <Card.Body>
                    <Card.Title className="text-truncate">{ child_care.operation_name }</Card.Title>
                    <Card.Text><b>Address:</b> { child_care.location_address }</Card.Text>
                    <Card.Text><b>County:</b> { child_care.county }</Card.Text>
                    <Card.Text><b>Days of Operation:</b> { child_care.days_of_operation }</Card.Text>
                    <Card.Text><b>Hours of Operation:</b> { child_care.hours_of_operation }</Card.Text>
                    <Card.Text><b>Ages Served:</b> { ages }</Card.Text>
                </Card.Body>
            </Card>
        </Link>
    )
};

export default ChildCareGrid;