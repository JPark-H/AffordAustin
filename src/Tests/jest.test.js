import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import About from "../Components/Pages/About/About";
import Housing from "../Components/Pages/InstancePages/Housing";
import ChildCare from "../Components/Pages/InstancePages/ChildCare";
import Jobs from "../Components/Pages/InstancePages/Jobs";

import Navbar from "../Components/MainNavBar/MainNavBar";

import ChildCareGrid from "../Components/Pages/GridPages/ChildCareGrid";
import HousingGrid from "../Components/Pages/GridPages/HousingGrid";
import JobGrid from "../Components/Pages/GridPages/JobGrid";

configure({ adapter: new Adapter() });

describe("Render components", () => {
  it("1. render About page", () => {
    const tree = shallow(<About />);
    expect(tree).toMatchSnapshot();
  });

  it("2. Housing should exist", () => {
    expect(Housing).toBeDefined();
  });

  it("3. Housing matches snapshot", () => {
    const tree = shallow(<Housing />);
    expect(tree).toMatchSnapshot();
  });

  it("4. ChildCare should exist", () => {
    expect(ChildCare).toBeDefined();
  });

  it("5. ChildCare matches snapshot", () => {
    const tree = shallow(<ChildCare />);
    expect(tree).toMatchSnapshot();
  });

  it("6. Jobs should exist", () => {
    expect(Jobs).toBeDefined();
  });

  it("7. Jobs matches snapshot", () => {
    const tree = shallow(<Jobs />);
    expect(tree).toMatchSnapshot();
  });

  it("8. Navbar exists and matches snapshot", () => {
    const navbar = shallow(<Navbar />);
    expect(navbar).toBeDefined;
    expect(navbar).toMatchSnapshot();
  });

  it("9. ChildCareGrid matches snapshot", () => {
    const ccgrid = shallow(<ChildCareGrid />);
    expect(ccgrid).toMatchSnapshot();
  });

  it("10. HousingGrid matches snapshot", () => {
    const ccgrid = shallow(<HousingGrid />);
    expect(ccgrid).toMatchSnapshot();
  });

  it("11. JobGrid matches snapshot", () => {
    const ccgrid = shallow(<JobGrid />);
    expect(ccgrid).toMatchSnapshot();
  });
});
