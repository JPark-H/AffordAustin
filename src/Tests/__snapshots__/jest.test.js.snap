// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components 1. render About page 1`] = `ShallowWrapper {}`;

exports[`Render components 3. Housing matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 5. ChildCare matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 7. Jobs matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 8. Navbar exists and matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 9. ChildCareGrid matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 10. HousingGrid matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 11. JobGrid matches snapshot 1`] = `ShallowWrapper {}`;
